### PR TITLE
kvm: make UEFI host check to support both Ubuntu and EL

### PR DESCRIPTION
**Pull request by rohityadavcloud**
Small fix to also support UEFI check on agent connect. Refer: https://github.com/apache/cloudstack/pull/6139/files

### CODE REVIEW

**Comment by rohityadavcloud** [comment]
Didn't test it myself, based on some code review.
@blueorangutan package

**Comment by rohityadavcloud** [comment]
@blueorangutan package

**Comment by rohityadavcloud** [comment]
@blueorangutan test

**Comment by rohityadavcloud** [comment]
@blueorangutan help

**Review by weizhouapache — APPROVED** [review]
code lgtm

Need minor change on trillian so we can test the UEFI feature on ubuntu oses.

**Comment by rohityadavcloud** [comment]
@blueorangutan test ubuntu20 kvm-ubuntu20

**Comment by rohityadavcloud** [comment]
@blueorangutan test ubuntu20 kvm-ubuntu20

File: plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtReadyCommandWrapper.java [hunk 1]
```diff
@@ -42,18 +42,21 @@ public final class LibvirtReadyCommandWrapper extends CommandWrapper<ReadyComman
...
 
         return new ReadyAnswer(command, hostDetails);
     }
 
-    private boolean hostSupportsUefi() {
+    private boolean hostSupportsUefi(boolean isUbuntuHost) {
         String cmd = "rpm -qa | grep -i ovmf";
+        if (isUbuntuHost) {
+            cmd = "dpkg -l ovmf";
+        }
```
Comment (DaanHoogland) [inline plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtReadyCommandWrapper.java:56]:
```suggestion
        String cmd = null;
        if (isUbuntuHost) {
            cmd = "dpkg -l ovmf";
        } else {
            cmd = "rpm -qa | grep -i ovmf";
        }
```

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Comment by DaanHoogland** [comment]
@rohityadavcloud, will this require extra testing?
Will you apply my suggestion?

File: plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtReadyCommandWrapper.java (hunk 1 above)
Comment (rohityadavcloud) [reply plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtReadyCommandWrapper.java:56]: I prefer to not have the cmd set to null when we know that's the default case.

**Comment by rohityadavcloud** [comment]
@DaanHoogland only if you insist, I prefer to set cmd to the default case (and not null; just the style to having less branching read better to me). I think it may not need additional testing; what we're doing is to enable the case when Ubuntu based KVM hosts. I've done testing of the pkg name and dpkg -l output manually to verify that it returns non-zero exit code when the pkg is not installed.

**Comment by DaanHoogland** [comment]
> @DaanHoogland only if you insist, I prefer to set cmd to the default case (and not null; just the style to having less branching read better to me). I think it may not need additional testing; what we're doing is to enable the case when Ubuntu based KVM hosts. I've done testing of the pkg name and dpkg -l output manually to verify that it returns non-zero exit code when the pkg is not installed.

Not insisting, Ilike @weizhouapache 's suggestion better anyway ;)

**Review by shwstppr — APPROVED** [review]

**Pull request merged**